libsolidity/lsp/ReferenceCollector.h

```diff
@@ -0,0 +1,69 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsolidity/ast/ASTForward.h>
+#include <libsolidity/ast/ASTVisitor.h>
+
+namespace solidity::lsp
+{
+
+enum class DocumentHighlightKind
+{
+	Unspecified = 0,    //!< could be for example a highlight found in a comment
+	Text = 1,           //!< a textual occurrence
+	Read = 2,           //!< read access to a variable
+	Write = 3,          //!< write access to a variable
+};
+
+// Represents a symbol / AST node that is to be highlighted, with some context associated.
+using Reference = std::tuple<langutil::SourceLocation, DocumentHighlightKind>;
+
+class ReferenceCollector: public frontend::ASTConstVisitor
+{
+public:
+	ReferenceCollector(frontend::Declaration const& _declaration, std::string const& _sourceIdentifierName);
```

Comment:
Looks like the `_sourceIdentifierName` is optional?
```suggestion
	ReferenceCollector(frontend::Declaration const& _declaration, std::string const& _sourceIdentifierName = "");
```

Also, it really needs some docs saying in what situations it should be specified. When I started the review I thought it was just because it was easier for you to get `Declaration::name()` from identifiers, paths, etc. and pass it through a parameter. But now I see that it's more likely that it's due to the fact that declarations can be renamed in imports?